### PR TITLE
Removed reference to older VelocyPackDumper

### PR DESCRIPTION
**Pull request by ignacionr**
### Scope & Purpose

We want to get rid of older direct inclusion of VelocyPack, in favor of its separate current directory; the single usage of the older VelocyPack dumper was detected and removed from VstResponse, a unit test was created.

*Dropping an older reference to VelocyPackDumper*

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [x] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: lib/Rest/VstResponse.cpp [hunk 1]
```diff
@@ -74,7 +73,7 @@ void VstResponse::addPayload(VPackSlice const& slice,
       VPackBuffer<uint8_t> tmpBuffer;
       tmpBuffer.reserve(slice.byteSize());  // reserve space already
       VPackBuilder builder(tmpBuffer, options);
-      VelocyPackHelper::sanitizeNonClientTypes(slice, VPackSlice::noneSlice(),
+       VelocyPackHelper::sanitizeNonClientTypes(slice, VPackSlice::noneSlice(),
```
Comment (jsteemann) [inline lib/Rest/VstResponse.cpp:76]: unrelated whitespace change, which we can revert
```suggestion
      VelocyPackHelper::sanitizeNonClientTypes(slice, VPackSlice::noneSlice(),
```

File: tests/RestHandler/VstResponse-test.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// --SECTION--                                                 setup / tear-down
+// -----------------------------------------------------------------------------
+
+class VstResponseTest
+    : public ::testing::Test {
+ protected:
+
+  VstResponseTest() {
+  }
+
```
Comment (jsteemann) [inline tests/RestHandler/VstResponse-test.cpp:42]: it looks like this is not needed and can be removed
```suggestion
```

File: tests/RestHandler/VstResponse-test.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#include "Rest/VstResponse.h"
+#include "velocypack/Parser.h"
+#include "Basics/encoding.h"
+
+#define ASSERT_ARANGO_OK(x) \
+  { ASSERT_TRUE(x.ok()) << x.errorMessage(); }
+
+// -----------------------------------------------------------------------------
+// --SECTION--                                                 setup / tear-down
+// -----------------------------------------------------------------------------
```
Comment (jsteemann) [inline tests/RestHandler/VstResponse-test.cpp:34]: this seems to be unused in this test, so I suggest removing it
```suggestion
#define ASSERT_ARANGO_OK(x) \
  { ASSERT_TRUE(x.ok()) << x.errorMessage(); }

// -----------------------------------------------------------------------------
// --SECTION--                                                 setup / tear-down
// -----------------------------------------------------------------------------
```

File: tests/RestHandler/VstResponse-test.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  // add the payload
+  resp.addPayload(slice, nullptr, true);
+
+  // retrieve the payload
+  auto const payload {resp.payload()};
+
+  EXPECT_EQ(payload.length(), 1);
+  EXPECT_EQ(payload.byteSize(), 1);
+  EXPECT_EQ(payload[0], '3');
```
[the remark this answers is not in the document]
Comment (ignacionr) [reply tests/RestHandler/VstResponse-test.cpp:74]: This explanation is very welcome. Actually me too I was under the impression that not getting the inserted numeric value was plain wrong. Now with a much better picture, I may be able to create the correct test, in addition to the integration. Thank you for the through review, it helps in many aspects.

File: lib/Rest/VstResponse.cpp (hunk 1 above)
Comment (ignacionr) [reply lib/Rest/VstResponse.cpp:76]: Added.

File: tests/RestHandler/VstResponse-test.cpp (hunk 2 above)
Comment (ignacionr) [reply tests/RestHandler/VstResponse-test.cpp:42]: Removed.

File: tests/RestHandler/VstResponse-test.cpp (hunk 3 above)
Comment (ignacionr) [reply tests/RestHandler/VstResponse-test.cpp:34]: Removed as suggested. Thank you.

File: tests/RestHandler/VstResponse-test.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (ignacionr) [reply tests/RestHandler/VstResponse-test.cpp:74]: Modified the test as discussed. Thank you!

**Review by jsteemann — APPROVED** [review]
